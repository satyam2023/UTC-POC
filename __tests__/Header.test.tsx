
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import Header from '../src/components/AppHeader';

describe('Header Component', () => {

  test("Component",()=>{
    const root=render(<Header topheading="Test Heading" />);
    expect(root.toJSON()).toMatchSnapshot();
  })

  it('renders correctly', () => {
    const { getByText, queryByText ,} = render(<Header topheading="Test Heading" />);
    expect(getByText('Test Heading')).toHaveLength;
  });
 

  it('renders logout button when isLogoutButton is true', () => {
    const { getByText } = render(<Header topheading="Test Heading" isLogoutButton={true} />);
    expect(getByText('LogOUt')).toBeTruthy();
  });

  it('calls rightButtonPress when the logout button is pressed', () => {
    const mockRightButtonPress = jest.fn();
    const { getByText } = render(
      <Header topheading="Test Heading" isLogoutButton={true} rightButtonPress={mockRightButtonPress} />
    );

    fireEvent.press(getByText('LogOUt'));
    expect(mockRightButtonPress).toHaveBeenCalled();
  });
});
