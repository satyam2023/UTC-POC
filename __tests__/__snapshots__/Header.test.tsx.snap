// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "blue",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
  testID="appHeaderContainer"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    />
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "500",
          "marginLeft": 16,
        }
      }
    >
      Test Heading
    </Text>
  </View>
</View>
`;
